### ISSUE TITLE
feat: add datasource for pagerduty and slack integration

### ISSUE THREAD

**Issue by skang0601** [DataDog/terraform-provider-datadog#1718]
### Terraform Version

`1.3.7`

### Affected Resource(s)

Please list the resources as a list, for example:

-   datadog_integration_pagerduty
-   datadog_integration_slack

### Expected Behavior

It would be nice to have a data source for these two integration types to query for their existence behavior in attempting to create the same integration again. This is especially helpful since these are a global integration and multiple different terraform project/codebases may want to reference the same notification name.
 
### Actual Behavior

Currently, no data sources exist for these integration.

**Comment by eugene-deng** [comment]
Hi - is there any plan for this request?
We are creating integrations for multiple slack channel but some are already created. It is all or nothing situation for us now and it is very painful as we don't even own these integrations.

**Comment by barryoneill** [comment]
Yes - same here.  Especially since @scemily13 has done the work for you :)
